pkg/controller/clone/planner.go

```diff
@@ -553,3 +759,46 @@ func createDesiredClaim(namespace string, targetClaim *corev1.PersistentVolumeCl
 
 	return desiredClaim
 }
+
+func createTempSourceClaim(ctx context.Context, namespace string, targetClaim *corev1.PersistentVolumeClaim, snapshot *snapshotv1.VolumeSnapshot, client client.Client) (*corev1.PersistentVolumeClaim, error) {
+	// Attempting to get a storageClass compatible with the source snapshot, but different from the one used in the target claim.
+	// This allows us to be consistent with the host-assisted clone behavior.
+	// If unable to do so, we'll use the target Claim one anyway.
+	scName := *targetClaim.Spec.StorageClassName
+	if snapshot.Status == nil || snapshot.Status.BoundVolumeSnapshotContentName == nil {
+		return nil, fmt.Errorf("volumeSnapshotContent name not found")
+	}
+	volumeSnapshotContent := &snapshotv1.VolumeSnapshotContent{}
+	if err := client.Get(ctx, types.NamespacedName{Name: *snapshot.Status.BoundVolumeSnapshotContentName}, volumeSnapshotContent); err != nil {
+		return nil, err
+	}
+	var matches []string
+	storageClasses := &storagev1.StorageClassList{}
+	if err := client.List(ctx, storageClasses); err != nil {
+		return nil, err
+	}
+	for _, storageClass := range storageClasses.Items {
+		if storageClass.Provisioner == volumeSnapshotContent.Spec.Driver && storageClass.Name != scName {
+			matches = append(matches, storageClass.Name)
+		}
+	}
+	if len(matches) > 0 {
+		sort.Strings(matches)
+		scName = matches[0]
+	}
```

Comment:
why not break out getting scName into a separate function?

Reply:
also, should return an error if scName cannot be found.  it is dereferenced later

Reply:
My idea was to use the target claim storage class if no match was found. The clone still seemed to work in that case.

Reply:
No way that can be guaranteed to work.  If it does then the "snapshot" clone method should have been chosen by the planner.  Sounds like you explicitly specified the "copy" clone type when creating the target PVC?

Reply:
The target storage class should not be considered when choosing.  It is possible that the chosen sc is the same as the target but that is a coincidence